app/src/main/java/com/kevinlu/watstats/MainActivity.java

```diff
@@ -1,13 +1,210 @@
 package com.kevinlu.watstats;
 
-import android.os.Bundle;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.viewpager.widget.ViewPager;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.os.Handler;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.francochen.watcard.WatCardAPI;
+import com.francochen.watcard.WatCardService;
+import com.francochen.watcard.model.RequestVerificationToken;
+import com.francochen.watcard.model.balance.BalanceType;
+import com.francochen.watcard.model.balance.Balances;
+import com.google.android.material.tabs.TabLayout;
+import com.kevinlu.watstats.models.AccountBalance;
+import com.kevinlu.watstats.models.MonthlySpending;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
 
+    TextView totalBalance;
+    ViewPager accountBalances;
+    ViewPager monthlySpendings;
+    AccountBalanceAdapter accountBalanceAdapter;
+    MonthlySpendAdapter monthlySpendAdapter;
+    List<AccountBalance> accountBalanceList;
+    List<MonthlySpending> monthlySpendingList;
+    WatCardAPI api = new WatCardAPI();
+    WatCardService service = api.createService();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        totalBalance = findViewById(R.id.total_balance);
+
+        accountBalanceList = new ArrayList<>();
+
+        accountBalanceAdapter = new AccountBalanceAdapter(accountBalanceList, this);
+
+        accountBalances = findViewById(R.id.account_balances);
+        accountBalances.setAdapter(accountBalanceAdapter);
+        accountBalances.setPadding(dpToPx(36), dpToPx(36), dpToPx(205), 0);
+
+        accountBalances.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                if (position < accountBalanceList.size() - 1) {
+                    return;
+                }
+                accountBalances.setCurrentItem(position - 1, true);
+            }
+
+            @Override
+            public void onPageScrollStateChanged(int state) {
+
+            }
+        });
```

Comment:
Since you only require overriding onPageSelected, use SimpleOnPageChangeListener instead.